File: windows/Liferay.SDK/Service/V73/Commercetaxmethod/CommercetaxmethodService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercetaxmethod
{
	public class CommercetaxmethodService : ServiceBase
	{
		public CommercetaxmethodService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> SetActiveAsync(long commerceTaxMethodId, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/set-active", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceTaxMethodAsync(IDictionary<string, string> nameMap, IDictionary<string, string> descriptionMap, string engineKey, bool percentage, bool active, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("nameMap", nameMap);
			_parameters.Add("descriptionMap", descriptionMap);
			_parameters.Add("engineKey", engineKey);
			_parameters.Add("percentage", percentage);
			_parameters.Add("active", active);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/add-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceTaxMethodAsync(long userId, long groupId, IDictionary<string, string> nameMap, IDictionary<string, string> descriptionMap, string engineKey, bool percentage, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("groupId", groupId);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("descriptionMap", descriptionMap);
			_parameters.Add("engineKey", engineKey);
			_parameters.Add("percentage", percentage);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/add-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> CreateCommerceTaxMethodAsync(long groupId, long commerceTaxMethodId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/create-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceTaxMethodAsync(long commerceTaxMethodId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/get-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchCommerceTaxMethodAsync(long groupId, string engineKey)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("engineKey", engineKey);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/fetch-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceTaxMethodsAsync(long groupId, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/get-commerce-tax-methods", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceTaxMethodsAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/get-commerce-tax-methods", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<dynamic> UpdateCommerceTaxMethodAsync(long commerceTaxMethodId, IDictionary<string, string> nameMap, IDictionary<string, string> descriptionMap, bool percentage, bool active)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);
			_parameters.Add("nameMap", nameMap);
			_parameters.Add("descriptionMap", descriptionMap);
			_parameters.Add("percentage", percentage);
			_parameters.Add("active", active);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/update-commerce-tax-method", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceTaxMethodAsync(long commerceTaxMethodId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxmethod/delete-commerce-tax-method", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercetaxmethod/CommercetaxmethodService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercetaxmethod;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercetaxmethodService extends BaseService {

	public CommercetaxmethodService(Session session) {
		super(session);
	}

	public JSONObject addCommerceTaxMethod(JSONObject nameMap, JSONObject descriptionMap, String engineKey, boolean percentage, boolean active, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("descriptionMap", checkNull(descriptionMap));
			_params.put("engineKey", checkNull(engineKey));
			_params.put("nameMap", checkNull(nameMap));
			_params.put("percentage", percentage);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercetaxmethod/add-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addCommerceTaxMethod(long userId, long groupId, JSONObject nameMap, JSONObject descriptionMap, String engineKey, boolean percentage, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("descriptionMap", checkNull(descriptionMap));
			_params.put("engineKey", checkNull(engineKey));
			_params.put("groupId", groupId);
			_params.put("nameMap", checkNull(nameMap));
			_params.put("percentage", percentage);
			_params.put("userId", userId);

			_command.put("/commerce.commercetaxmethod/add-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject createCommerceTaxMethod(long groupId, long commerceTaxMethodId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("groupId", groupId);

			_command.put("/commerce.commercetaxmethod/create-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceTaxMethod(long commerceTaxMethodId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxMethodId", commerceTaxMethodId);

			_command.put("/commerce.commercetaxmethod/delete-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchCommerceTaxMethod(long groupId, String engineKey) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("engineKey", checkNull(engineKey));
			_params.put("groupId", groupId);

			_command.put("/commerce.commercetaxmethod/fetch-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCommerceTaxMethod(long commerceTaxMethodId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxMethodId", commerceTaxMethodId);

			_command.put("/commerce.commercetaxmethod/get-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceTaxMethods(long groupId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);

			_command.put("/commerce.commercetaxmethod/get-commerce-tax-methods", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceTaxMethods(long groupId, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("groupId", groupId);

			_command.put("/commerce.commercetaxmethod/get-commerce-tax-methods", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONObject setActive(long commerceTaxMethodId, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceTaxMethodId", commerceTaxMethodId);

			_command.put("/commerce.commercetaxmethod/set-active", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceTaxMethod(long commerceTaxMethodId, JSONObject nameMap, JSONObject descriptionMap, boolean percentage, boolean active) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("active", active);
			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("descriptionMap", checkNull(descriptionMap));
			_params.put("nameMap", checkNull(nameMap));
			_params.put("percentage", percentage);

			_command.put("/commerce.commercetaxmethod/update-commerce-tax-method", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}